### ISSUE TITLE
Need a solution for custom drawing view in an ORKQuestionStep

### ISSUE THREAD

**Issue by berenyiaustin** [ResearchKit/ResearchKit#1140]
I've read every answer I can find for implementing custom views for an `ORKQuestionStep`, but I can't seem to find help. I'm creating an app for a physical therapy clinic that will replace all their patient paper forms with a survey on their iPad. I have all the questions created except for one that requires something outside the scope of the module. I'm trying to implement a question like this:

![bodydiagram](https://user-images.githubusercontent.com/18562453/43118977-c8653b92-8ed1-11e8-9fc0-b17dcf995da9.png)

A body diagram would be presented, and the user could draw a shape using `bezierPath`. When the user presses continue the image (and the `bezierPath`) would be saved as a result.

I need to accomplish this, but I don't know where to start. Any help or direction would be appreciated.

**Comment by ninoguba** [comment]
I will need this feature in my next project so will try to implement something easy.  I would likely expand the feature I implement in #1016

**Comment by davwillev** [comment]
Hi @ninoguba and @berenyiaustin  
Do you think it is worth trying to use a more standardised body/manikin maps in ResearhKit? [Another open source project](https://github.com/ctsit/imagemap) has utilised these body maps, which have the advantage of being gender sensitive:
![image](https://user-images.githubusercontent.com/35876659/43161275-9107ff3c-8f7f-11e8-838c-121393d8c287.png)
![image](https://user-images.githubusercontent.com/35876659/43161285-978b09da-8f7f-11e8-840a-866df9ac8929.png)

**Comment by srinathtm-apple** [comment]
Hey @berenyiaustin , you should look into ORKFreeHandDrawingView which we added just last month. This should handle all user touches and return the image as well the array of UIBezierPaths. We leveraged this view in ORKPDFViewerStep and ORKAmslerGridStep which you can check out as well to see how this can be used.

**Comment by ninoguba** [comment]
@berenyiaustin to make it simple I will go with "tap the areas of your body..." instead of letting them draw arbitrary shapes over the image.  Since I have worked on #1016 it looks much easier for me to expand from it.

@davwillev yes I would like to use standardized body maps like the one you've shared.  Will explore CHOIR's licensing verbiage...

**Comment by davwillev** [comment]
These regional maps work well within the REDCap system. I use the regional responses with 'branching logic' to then ask about the pain in each chosen region.
Obviously, I would imagine that only one side of the body (front or back) can be used per screen; not sure what the best order of this would be though.

**Comment by davwillev** [comment]
Certainly, in the pain literature, there have been a variety of regions defined for the body manikins. These CHOIR body regions are pretty good though, with meaningful anatomical regions, and given that they are already being used elsewhere (i.e. REDCap), I would be happy to see them implemented here.
I think it would be good to have the option to modify the images relatively easily though; there must be lots of use cases for choosing regions from images, so modifiability should be built in.

**Comment by ninoguba** [comment]
And this is why I'm more leaning toward my ORKImageChoiceAnswerFormat so you can provide your own images.  I just have to figure out positioning the images in a variable grid-like manner.

**Comment by davwillev** [comment]
I agree. Could the code be written to recognise certain tints in the image, bounded by a border with a specified tint (e.g. black)? This would enable regions in any shape of image to be recognised as long as the basic topgraphy was the same

**Comment by ninoguba** [comment]
@davwillev no, my approach will require you to splice up the body map image into separate smaller images so they would be easily adaptable to the current implementation of ORKImageChoiceAnswerFormat.  It will take me a bit of time to implement this so I'll just post an update when ready.

**Comment by berenyiaustin** [comment]
Hi, @ninoguba :) Has there been any progress with this feature? I know it must be a heavy undertaking.

**Comment by ninoguba** [comment]
@berenyiaustin requirement priorities changed and this one got dropped so I'm not working on this feature implementation at the moment.  Sorry.

**Comment by Pariecemckinney-apple** [comment]
This feature request has been recorded and will be tracked internally. Please feel free to open another issue if further discussion is needed.
